### ISSUE TITLE
[Bug]: Exception getting DISCO information

### ISSUE THREAD

**Issue by InteXX** [CoreWCF/CoreWCF#1509]
### Duplicate ?

- [X] I have searched issues/discussions and did not find other issues/discussions reporting this bug.

### Product version

1.6.0

### Describe expected behavior

A well-formed XML response providing the service's WSDL URL, as is available in a `net48` service.

### Describe actual behavior

```
An ExceptionDetail, likely created by IncludeExceptionDetailInFaults=true, whose value is:
System.InvalidOperationException: Token StartDocument in state Document would result in an invalid XML document.
   at System.Xml.XmlWellFormedWriter.ThrowInvalidStateTransition(Token token, State currentState)
   at System.Xml.XmlWellFormedWriter.AdvanceState(Token token)
   at System.Xml.XmlWellFormedWriter.WriteStartDocumentImpl(XmlStandalone standalone)
   at CoreWCF.Description.ServiceMetadataExtension.HttpGetImpl.DiscoResult.Write(XmlWriter writer)
   at CoreWCF.Description.ServiceMetadataExtension.HttpGetImpl.MetadataResult.WriteResponseAsync(HttpResponse response)
   at CoreWCF.Description.ServiceMetadataExtension.HttpGetImpl.TryHandleMetadataRequestAsync(HttpContext requestContext, IQueryCollection queries)
   at CoreWCF.Description.ServiceMetadataExtension.HttpGetImpl.ProcessHttpRequest(HttpContext requestContext)
   at CoreWCF.Description.ServiceMetadataExtension.HttpGetImpl.HandleRequest(HttpContext httpContext)
```

### Which binding

BasicHttp

### security

Transport

### Which .NET version

.NET 7

### Which os platform

Windows

### Code snippet used to reproduce the issue

```c#
Browse to https://example.com/service?disco (as indicated in the HTML source of https://example.com/service, within the <head /> element).
```

### Stacktrace if any

```shell
See above.
```

**Comment by stefannikolei** [comment]
@InteXX Do you have a minimal repro code? With the Information you initially wrote in the Issue no one can do anything with it.

**Comment by InteXX** [comment]
> no one can do anything with it

Oops, I beg to differ.

The steps (step, actually) to reproduce it are in the Code Snippet section.

The CoreWCF [sample project](https://github.com/CoreWCF/samples/tree/main/Basic/Binding/Basic/Http/Service) is sufficient to demonstrate the bug.

**Comment by stefannikolei** [comment]
> > no one can do anything with it
> 
> Oops, I beg to differ.
> 
> The steps (step, actually) to reproduce it are in the Code Snippet section.
> 
> The CoreWCF [sample project](https://github.com/CoreWCF/samples/tree/main/Basic/Binding/Basic/Http/Service) is sufficient to demonstrate the bug.

Oh you try to get the WSDL with ?disco ? --> >Try it with http://localhost:500/CalculatorService/basicHttp?wsdl

**Comment by InteXX** [comment]
> Oh you try to get the WSDL with ?disco

Actually, no. I'm trying to get the WSDL URL from the service URL, as in a `net48` service.

In a `net48` service, you can:

1. Browse to the main service URL
2. Inspect the HTML source
3. Browse to the DISCO URL
4. Get the WSDL URL

This (or something similar) is the sequence of steps that a WCF code generator takes in order to get the WSDL URL.

For example, connect to the service URL of a `net48` service using the WCF Test Client that's bundled with Visual Studio (e.g. http://example.com/service). It works as expected.

Now try to connect to the service URL of a CoreWCF service. You'll get the above exception. We must include the `?WSDL` query string.

I struggled with this while getting CoreWCF running for the first time. Because of the exception I assumed there was something wrong with my code, and I lost some time chasing down rabbit holes. I assumed that my new CoreWCF service would behave during the Add Connection experience the same as a `net48` service does.

Thus my report.

**Comment by stefannikolei** [comment]
Ah ok ty. Now i know what you tried. I personally never did it that way I always passed the Wsdl uri. 

Perhaps I find time to debug it 😇

**Comment by InteXX** [comment]
Goodness, that was quick!

I didn't realize the fix was that easy, or I could've saved you the trouble.

**Comment by InteXX** [comment]
@mconnew 

Thanks!

**Comment by mconnew** [comment]
I don't normally accept PR's without a test which exercises the code, but I made an exception in this case as the error call stack is very obvious that the is the right one. I did open an issue for a unit test to be created for this if anyone wants to add coverage (#1516), but don't feel obligated. I just finished up on a massive chunk of WCF Client work so can now shift gears to CoreWCF and will release a new version very soon.
